Formula/lsyncd.rb

```diff
@@ -69,10 +69,10 @@ class Lsyncd < Formula
       "12.3"    => ["xnu-8020.101.4.tar.gz",      "df715e7b2bd5db0ba212b5b0613fbbc85c3cbc4e61f6ee355a8b6cf9a87d3374"],
     }
 
-    macos_version = if MacOS.version >= :big_sur
-      MacOS.full_version.major_minor # Ignore bugfix/security updates
-    else
-      MacOS.full_version
+    macos_version = MacOS.full_version.major_minor # Ignore bugfix/security updates
+
+    on_catalina :or_older do
+      macos_version = MacOS.full_version
```

Comment:
Feels like a Homebrew/brew API for this might be nice if it's used anywhere else.

Reply:
I think it's used in only two spots. Here, and inside a `resource` in [`go`](https://github.com/Homebrew/homebrew-core/pull/105149/files/b3fe2b32e1be649387c520e294ba1a5a838dbc28#diff-2b4628657de5dd65ff1afe2817f23c4deb13e2ba587b1fe8a7605479c5e0c660)

Reply:
However, this kind of pattern where you assign variables and then interpolate them in URLs is [_heavily_ used in casks](https://github.com/Homebrew/brew/pull/13451#issuecomment-1165232527), so we'll have to address this at some point, I think.

When we make a final decision for what to do with casks, I think we can just mimic that here.

Reply:
I think it'd be nicer for casks to look a bit more like formulae, as time goes on, but I don't think that's blocking or needs done now.